lib/elixir/lib/module.ex

```diff
@@ -1789,6 +1789,22 @@ defmodule Module do
     :ets.insert(set, {key, value, line})
   end
 
+  defp put_attribute(module, :on_load, value, line, _set, bag) do
+    try do
+      :ets.lookup_element(bag, {:accumulate, :on_load}, 2)
+    catch
+      :error, :badarg -> :ets.insert(bag, {{:accumulate, :on_load}, value})
+    else
+      _ ->
+        [{_, _, _, file: file, line: line}] = attribute_stack(module, line)
+
+        raise CompileError,
```

Comment:
Nowhere in this module we raise a compile error, so I am thinking a `raise ArgumentError` is enough as it will contain relevant stacktrace info.